### ISSUE TITLE
Manage stale branches

### ISSUE THREAD

**Issue by johnd0e** [Leaflet/Leaflet#7032]
Leaflet repo currently contains 96 branches, most of them are [stale](https://github.com/Leaflet/Leaflet/branches/stale).

So I propose:
- [ ] Remove merged
- [ ] Remove closed
- [ ] Where related PR absent, open one, with some description

Thoughts?

**Comment by johnd0e** [comment]
## Merged:
- #6748 `plugin-maidenhead`
Updated 8 months ago by @IvanSanchez
- #5859 `fix-circle-bounds-calc`
Updated 2 years ago by Per Liedman
- #5690 `fix-paused-drag-inertia`
Updated 3 years ago by Per Liedman
- #5689 `warn-adding-non-layers`
Updated 3 years ago by Per Liedman
- #5667 `isflat`
Updated 3 years ago by @yohanboniface
- #5577 `docsUpdateWhenIdle`
Updated 3 years ago by perliedman
- #5581 `1.1.0-blog-post`
Updated 3 years ago by Per Liedman
- #5580 `default-icon-regexp`
Updated 3 years ago by IvanSanchez
- #5574 `divIconContent`
Updated 3 years ago by @ghybs
- #5555 `imageoverlay-classname`
Updated 3 years ago by @mourner
- #5572 `quick-start-map-id`
Updated 3 years ago by IvanSanchez
- #5507 `fix-marker-enter-key`
Updated 3 years ago by @perliedman
- #4883 `bubbly-option`
Updated 3 years ago by IvanSanchez
- #5498 `fix-canvas-empty-polyline`
Updated 3 years ago by perliedman
- #5476 `no-gap-hooks`
Updated 3 years ago by IvanSanchez
- #5480 `scroll-pixelratio`
Updated 3 years ago by IvanSanchez
- #5465 `only-rearrange-dom-when-needed`
Updated 3 years ago by Per Liedman
- #5404 `delayed-paths`
Updated 3 years ago by IvanSanchez
- #5007 `tutorial-zoom-delta`
Updated 3 years ago by perliedman
- #5378 `stop-map-on-drag-start`
Updated 3 years ago by IvanSanchez
- #5331 `keyboard-escape`
Updated 3 years ago by IvanSanchez
- #5318 `disable-click-prop-in-zoom-control`
Updated 3 years ago by Per Liedman
- #5303 `tap-highlight-color`
Updated 3 years ago by IvanSanchez
- #5157 `zero-bounds`
Updated 3 years ago by IvanSanchez
- #5280 `layers-control-scroll-2`
Updated 3 years ago by IvanSanchez
- #5274 `release-v1.0.3`
Updated 3 years ago by IvanSanchez
- #5166 `wraplatlngbounds`
Updated 3 years ago by IvanSanchez
- #5177 `infinite-tile-errors`
Updated 3 years ago by IvanSanchez
- #5100 `canvas-click-single-layer`
Updated 3 years ago by perliedman
- #4711 `control-layers-comparelayers`
Updated 3 years ago by Per Liedman
- #4916 `layers-max-zoom`
Updated 3 years ago by Per Liedman
- #5049 `as-feature-collection`
Updated 3 years ago by Per Liedman
- #5070 `attribution-on-all-layers`
Updated 3 years ago by Per Liedman
- #5054 `path-events-refactor`
Updated 3 years ago by IvanSanchez
- #5021 `dup-if`
Updated 4 years ago by yohanboniface
- #4555 `fix-removing-all-listeners`
Updated 4 years ago by perliedman
- #4543 `grid-layer-docs`
Updated 4 years ago by @nathancahill
- #4531 `wms-docs`
Updated 4 years ago by IvanSanchez
- #4520 `insecure-geolocation`
Updated 4 years ago by IvanSanchez
- #4513 `control-extension-docs`
Updated 4 years ago by nathancahill
- #4512 `correct-geojson-docs`
Updated 4 years ago by nathancahill

**Comment by johnd0e** [comment]
### Closed:
- #6691 `cssIconDefault`
Updated 2 years ago by ghybs
- #5290 `pixelgrid`
Updated 3 years ago by IvanSanchez
- #5279 `layers-control-scroll`
Updated 3 years ago by IvanSanchez
- #5249 `project-update-paths`
Updated 3 years ago by perliedman
- #5172 `canvas-reset-2`
Updated 3 years ago by IvanSanchez
- #5171 `canvas-reset`
Updated 3 years ago by IvanSanchez
- #5002 `rollup3`
Updated 4 years ago by IvanSanchez
- #4965 `tutorials-cleanup`
Updated 4 years ago by IvanSanchez
- #4947 `v1-blog-the-wait-is-over`
Updated 4 years ago by IvanSanchez
- #4894 `blog-foss4g`
Updated 4 years ago by IvanSanchez
- #4356 `gh-pages-extending`
Updated 4 years ago by IvanSanchez
- #4709 `event-refactor-round-2`
Updated 4 years ago by Per Liedman
- #4710 `control-layers-eachlayer`
Updated 4 years ago by yohanboniface
- #4649 `tile-prune-lru`
Updated 4 years ago by IvanSanchez
- #4614 `gridlayer-margin`
Updated 4 years ago by IvanSanchez
- #4516 `type-definitions`
Updated 4 years ago by nathancahill
- #4507 `mutation-observer`
Updated 4 years ago by IvanSanchez
- #6520 `0.7`
Updated 4 years ago by Christopher Green
- #4197 `slimerjs`
Updated 4 years ago by IvanSanchez
- #4074 `move-latlng-equals-to-crs`
Updated 4 years ago by perliedman
- #3809 `mobile-setview`
Updated 5 years ago by yohanboniface
- #3598 `geojson-round-trip-test`
Updated 5 years ago by @patrickarlt
- #3581 `scroll-prevent`
Updated 5 years ago by IvanSanchez
- #3528 `domevent-once`
Updated 5 years ago by yohanboniface

**Comment by johnd0e** [comment]
### Without related PR:

- [`propagate-marker-drag`](https://github.com/Leaflet/Leaflet/tree/propagate-marker-drag)
Updated 3 months ago by IvanSanchez
- [`pr/6021`](https://github.com/Leaflet/Leaflet/tree/pr/6021)
Updated 2 years ago by IvanSanchez
- [`fix-circle-while-zooming`](https://github.com/Leaflet/Leaflet/tree/fix-circle-while-zooming)
Updated 3 years ago by Per Liedman
- [`canvas-improvements`](https://github.com/Leaflet/Leaflet/tree/canvas-improvements)
Updated 3 years ago by Per Liedman
- [`ffcfcc1`](https://github.com/Leaflet/Leaflet/tree/ffcfcc1)
Updated 3 years ago by IvanSanchez
- [`rotate`](https://github.com/Leaflet/Leaflet/tree/rotate)
Updated 4 years ago by IvanSanchez
- [`rollup`](https://github.com/Leaflet/Leaflet/tree/rollup)
Updated 4 years ago by IvanSanchez
- [`split-css-autoprefixer`](https://github.com/Leaflet/Leaflet/tree/split-css-autoprefixer)
Updated 4 years ago by IvanSanchez
- [`gh-pages-custom-crs`](https://github.com/Leaflet/Leaflet/tree/gh-pages-custom-crs)
Updated 4 years ago by IvanSanchez
- [`drag-cancel-click`](https://github.com/Leaflet/Leaflet/tree/drag-cancel-click)
Updated 5 years ago by jfirebaugh
- [`cache-mouse-pos`](https://github.com/Leaflet/Leaflet/tree/cache-mouse-pos)
Updated 6 years ago by mourner

**Comment by mourner** [comment]
We should definitely remove merged branches. I actually just turned on the option to delete them automatically on merge, which will save us from those form now on.

For branches with closed PRs, I think they need a review — e.g. if the changes are substantial, sometimes they contain interesting attempts that may be useful in the future. If they are trivial, they should be deleted. If they are substantial but super old or addressing a part of the code that was significantly changed since then, they should be deleted too.

**Comment by mourner** [comment]
Did a pass and removed all merged branches and some unmerged ones with closed PRs — down to 46, but this definitely needs more cleanup.

**Comment by johnd0e** [comment]
> For branches with closed PRs, I think they need a review — e.g. if the changes are substantial, sometimes they contain interesting attempts that may be useful in the future.

If we find such case - then related PR should be reopened, right?
(Or create new PR with some description)

**Comment by mourner** [comment]
@johnd0e not necessarily. PRs should only be opened if there's an intention to merge. If there's just a branch with some code that _may_ be useful in future, we can keep it without a PR.

**Comment by johnd0e** [comment]
If not PR, then where we should discuss that branch?

When there is no intention to merge - PR can be opened in "draft" mode.

**Comment by jonkoops** [comment]
I haven't done any investigation, but this might help our cause: https://github.com/actions/stale

**Comment by Falke-Design** [comment]
I think this is exactly what was described [here](https://github.com/Leaflet/Leaflet/issues/7704#issuecomment-954545179) but I personally still think it's wrong to close the Issues automatical without interaction from a maintainer. There around 430 Issues and they can't get solved if the PRs don't get merged or if a decison is needed and no one decides it. A bot would close them all but the issues are still valid. I think this makes sense when we reduced the amount of issues and PRs manually and Leaflet is better maintained (which is currently good but we don't know what is in 2 months )
